apps/src/code-studio/initApp/loadApp.js

```diff
@@ -324,9 +328,18 @@ async function loadAppAsync(appOptions) {
     appOptions.exampleSolutions = exampleSolutions;
     appOptions.disableSocialShare = data.disableSocialShare;
 
-    // We do not need to process data.progress here because labs do not use
-    // the level progress data directly. (The progress bubbles in the header
-    // of the level pages are rendered by header.build in header.js.)
+    if (data.isStarted) {
+      appOptions.level.isStarted = data.isStarted;
+    }
+    if (data.skipInstructionsPopup) {
+      appOptions.level.skipInstructionsPopup = data.skipInstructionsPopup;
+    }
+    if (data.readonlyWorkspace) {
```

Comment:
Would it be possible to skip some of these checks and just directly assign the result of `data.readonlyWorkspace` to `appOptions.readonlyWorkspace` (and always return it from the endpoint)?

Reply:
When the user is signed-out, we only return `signedIn: false`.  Is your suggestion to return these values even when the user is signed-out?

Reply:
Oh right.. ok I take back my suggestion